components/vc/front/src/torque.c

```diff
@@ -120,6 +121,15 @@ static void torque_periodic_100Hz(void)
         }
     }
 
+#if FEATURE_IS_ENABLED(FEATURE_REVERSE)
+    CAN_gear_E direction = CAN_GEAR_FORWARD;
+    const bool direction_valid = CANRX_get_signal(VEH, STW_gear, &direction) == CANRX_MESSAGE_VALID;
+    if (direction_valid && (direction == CAN_GEAR_REVERSE))
+    {
+        torque_data.torque_request_max = DEFAULT_TORQUE_LIMIT_REVERSE;
+    }
```

Comment:
kinda meta and we can do this later, but what if instead of doing it this way we actually develop the concept of "gears"? Like, the vehicle starts in `GEAR_NEUTRAL` and then the steering wheel can submit "gear change requests" which the VC can then decide whether to honor or not.

That way, you can introduce additional checks in the VC and everything flows nicely.

So it would be like `STW_gearShiftRequest` which could be `GEAR_DRIVE`, `GEAR_NEUTRAL` or 	`GEAR_REVERSE`. Receive that in the VC, add whatever other checks you want (I can't think of any examples right now but I'm sure there are some) and then do the shift if the conditions are all good. Then you publish like a `VCREAR_gear` or something, and you can display that on the screen

Reply:
Lets chat and revisit in a follow up ticket. Might be a good thing to get a newer member on so they can get their feet wetter